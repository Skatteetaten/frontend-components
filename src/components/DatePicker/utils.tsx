export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
