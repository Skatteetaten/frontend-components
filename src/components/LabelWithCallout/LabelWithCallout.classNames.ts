import { FontSizes, FontWeights, PaletteProps } from '..';
import { mergeStyleSets } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { LabelWithCalloutProps } from './LabelWithCallout';

function getLabelSize(props: LabelWithCalloutProps) {
  switch (props.inputSize) {
    case 'small':
      return {
        fontSize: FontSizes.small
      };
    case 'large':
      return {
        fontSize: FontSizes.large
      };
    default:
      return {
        fontSize: FontSizes.medium
      };
  }
}

export const getClassNames = (props: LabelWithCalloutProps) => {
  const palette = getTheme().palette as PaletteProps;
  const { calloutFloating } = props;

  // @ts-ignore //TODO merge
  return mergeStyleSets({
    calloutContext: {
      selectors: {
        '& .ms-Callout-container': {
          position: !calloutFloating && 'inherit',
          margin: !calloutFloating ? '10px 0' : 0,
          width: '100%'
        },
        '& .ms-Callout': !calloutFloating && {
          left: '0 !important',
          top: '0 !important',
          position: 'inherit'
        },
        '& .ms-Callout-main': !calloutFloating && {
          maxWidth: '100%',
          display: 'inline-block'
        },
        '& .ms-Callout-beak': !calloutFloating && {
          left: '20px !important',
          top: '-8px !important'
        }
      }
    },
    label: {
      paddingBottom: 4,
      display: 'inline-block',
      selectors: {
        color: palette.skeColor.blackAlt,
        fontWeight: FontWeights.regular,
        ...getLabelSize(props),
        '& .ms-Label': {
          fontWeight: FontWeights.regular
        }
      }
    },
    labelAsLegend: {
      paddingLeft: 0,
      paddingBottom: 2,
      display: 'inline-block',
      selectors: {
        color: palette.skeColor.blackAlt,
        fontWeight: FontWeights.regular,
        ...getLabelSize(props),
        '& .ms-Label': {
          fontWeight: FontWeights.regular
        },

        '@supports (display: contents)': {
          display: 'contents'
        }
      }
    },
    labelIconArea: {
      height: '22px',
      display: 'inline-block',

      selectors: {
        '& button ': {
          top: '-1px',
          height: '22px'
        }
      }
    },
    icon: {
      color: palette.skeColor.blue,
      selectors: {
        '& i': {
          fontSize: 'large'
        },
        '&:focus&:after': {
          border: `2px solid ${palette.skeColor.blue}`,
          outline: 'none'
        }
      }
    },
    errorIcon: {
      position: 'absolute',
      left: -25,
      bottom: -30,
      selectors: {
        '& i': {
          color: palette.skeColor.pink,
          fontSize: 'large'
        }
      }
    }
  });
};
