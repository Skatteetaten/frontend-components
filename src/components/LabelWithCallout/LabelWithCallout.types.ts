import * as React from 'react';

import { CalloutProps } from '../Callout/Callout.types';

export enum calloutState {
  OPEN = 'OPEN',
  CLOSED = 'CLOSED',
}

export interface LabelWithCalloutProps
  extends React.HTMLAttributes<HTMLDivElement> {
  /** Id til legend/label */
  id?: string;
  /** Used for connecting to an input field */
  inputId?: string;
  /** Text to be shown in label */
  label?: string;
  /** aria-label */
  ariaLabel?: string;
  /** aria-label for help/warning button */
  buttonAriaLabel?: string;
  /** Title for help/warning button */
  buttonTitle?: string;
  /** If the Callout should float over page content */
  calloutFloating?: boolean;
  /** Custom classNames for Ã¥ overskrive styling */
  customClassNames?: {
    wrapper?: string;
    label?: string;
    legend?: string;
    helpicon?: string;
    helpPragraph?: string;
    warningicon?: string;
    warningPragraph?: string;
    editicon?: string;
    readonlyarea?: string;
    callout?: string;
  };
  editable?: boolean;
  editFunction?: () => void;
  help?: string | JSX.Element;
  /** When placed inside a fieldset, and the label element should be renderes as a legend element instead. */
  inFieldset?: boolean;
  inputSize?: 'small' | 'normal' | 'large';
  /** Event for callout **/
  onCalloutToggle?: (
    oldCalloutState: calloutState,
    newCalloutState: calloutState
  ) => void;
  /** CalloutProps som sendes videre ned til Callout */
  calloutProps?: CalloutProps;
  onRenderLabel?: any;
  readOnly?: boolean;
  warning?: string | JSX.Element;
}
