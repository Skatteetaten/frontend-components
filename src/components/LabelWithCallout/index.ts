export { default } from './LabelWithCallout';
export * from './LabelWithCallout';
