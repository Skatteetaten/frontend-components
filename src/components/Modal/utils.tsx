import { useEffect } from 'react';

export const createModalDomPlacement = (
  setFlag: (a: boolean) => void,
  shadowRootNode: Document | ShadowRoot | undefined
) => {
  if (!shadowRootNode) {
    setFlag(true);
    return;
  }

  const styleInjectorWrapperId = 'style-injector';
  const modalWrapperId = 'modal-wrapper';

  if (shadowRootNode.getElementById(modalWrapperId)) {
    setFlag(true);
  } else {
    const modalWrapper = document.createElement('div');
    modalWrapper.id = modalWrapperId;
    const styleInjectorWrapper = shadowRootNode.getElementById(
      styleInjectorWrapperId
    );
    const parent = styleInjectorWrapper ? styleInjectorWrapper : shadowRootNode;
    parent.appendChild(modalWrapper);
    setFlag(true);
  }
};

export const useEscOnPress = (
  ref: HTMLDivElement | null,
  onClose?: (ref: HTMLDivElement) => void
): void => {
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent): void => {
      if (ref == null) {
        return;
      }
      if (event.code === 'Escape' && onClose) {
        onClose(ref);
      }
    };

    if (ref) {
      window.addEventListener('keydown', onKeyDown as EventListener, false);
    }
    return function cleanup(): void {
      if (window) {
        window.removeEventListener('keydown', onKeyDown as EventListener);
      }
    };
  }, [ref, onClose]);
};
