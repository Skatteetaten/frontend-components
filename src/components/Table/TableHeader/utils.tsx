import { TFunction } from 'react-i18next';

export const getIconNameForTheadCell = (
  isSorted: boolean,
  isSortedAscending: boolean
) => {
  if (isSorted) {
    return isSortedAscending ? 'ArrowDown' : 'ArrowUp';
  }

  return 'ArrowUpDown';
};

export const getAriaLabelForTheadCell = (
  name: string | JSX.Element,
  fieldName: string,
  isSorted: boolean,
  isSortedAscending: boolean,
  t: TFunction<Array<string>>
) => {
  let ariaLabel = fieldName;
  if (typeof name === 'string') {
    ariaLabel = isSorted
      ? isSortedAscending
        ? name.concat(' ', t('table.sorted_ascending'))
        : name.concat(' ', t('table.sorted_descending'))
      : name.concat(' ', t('table.sortable'));
  }

  return ariaLabel;
};
