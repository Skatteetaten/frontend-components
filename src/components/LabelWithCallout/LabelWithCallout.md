**LabelWithCallout (Merkelapp med utropsboks): Brukes til å vise en merkelapp og tilhørende hjelpetekst**

```js
import { Button } from '@skatteetaten/frontend-components/Button';
import { LabelWithCallout } from '@skatteetaten/frontend-components/LabelWithCallout';

const [state, setState] = React.useState({ warning: false });

<div>
  <LabelWithCallout
    id={'lwc1'}
    label={'Omregistreringsavgift'}
    help={'Avgiften du må betale for å registrere kjøretøyet på en ny person.'}
    warning={
      state.warning && 'Du ser ut til å være fritatt for omregistreringsavgift.'
    }
    calloutProps={{ border: true }}
  />
  <br />
  <Button onClick={() => setState({ warning: !state.warning })}>
    {state.warning ? 'Vis med hjelpetekst' : 'Vis med varsel'}
  </Button>
</div>;
```

Brukt i kombinasjon med annen komponent, og veldig mye tekst.

```js
import { LabelWithCallout } from '@skatteetaten/frontend-components/LabelWithCallout';
import { SearchField } from '@skatteetaten/frontend-components/SearchField';

<div style={{ maxWidth: '400px' }}>
  <LabelWithCallout
    id={'lwc2'}
    label={'Søk'}
    help={
      'Her kan du søke etter personer og virksomheter. Dette søkefeltet er spesielt tilpasset for søk i befolkningen, og tar derfor hensyn til personvern. Det betyr at vi ikke kan vise noen resultater før du har bekreftet søket med enter, eller trykket på søkeikonet i feltet. Alle søk logges.'
    }
  />
  <SearchField
    searchFieldSize="standard"
    border="slim"
    placeholder="Skriv søkeord her"
    ariaLabel="Søkefelt"
  />
</div>;
```

Flytende, automatisk lukking og mye tekst:

```js
import { LabelWithCallout } from '@skatteetaten/frontend-components/LabelWithCallout';
import { SearchField } from '@skatteetaten/frontend-components/SearchField';

<div>
  <LabelWithCallout
    id={'lwc4'}
    label={'Søk'}
    help={
      <p style={{ width: '350px' }}>
        'Her kan du søke etter personer og virksomheter. Dette søkefeltet er
        spesielt tilpasset for søk i befolkningen, og tar derfor hensyn til
        personvern. Det betyr at vi ikke kan vise noen resultater før du har
        bekreftet søket med enter, eller trykket på søkeikonet i feltet. Alle
        søk logges. Her kan du IGJEN søke etter personer og virksomheter. Dette
        søkefeltet er spesielt tilpasset for søk i befolkningen, og tar derfor
        hensyn til personvern. Det betyr at vi ikke kan vise noen resultater før
        du har bekreftet søket med enter, eller trykket på søkeikonet i feltet.
        Alle søk logges.'
      </p>
    }
    calloutProps={{ autoDismiss: true }}
    calloutFloating
  />
</div>;
```

LabelWithCallout brukt i tekstfelt med automatisk lukking av utropsboksen:

```js
import { TextField } from '@skatteetaten/frontend-components/TextField';

const [state, setState] = React.useState({ value1: '' });

<>
  <div style={{ maxWidth: '400px' }}>
    <TextField
      label="Fullt navn"
      value={state.value1}
      onChange={(e, value) => setState({ value1: value })}
      help="Vi trenger å vite navnet ditt dersom vi skal kontakte deg senere."
      labelWithCalloutProps={{
        calloutProps: { autoDismiss: true },
      }}
    />
  </div>
</>;
```

```js noeditor beskrivelse
<>
  <h3>Hjelpekomponent</h3>
  <p>
    Dette er i hovedsak en hjelpekomponent som brukes av input-komponentene. Det
    er for eksempel denne som brukes til å vise label og hjelpetekst og i toppen
    av TextField og DatePicker.
  </p>
  <p>
    Bruk denne komponenten i tilfeller hvor det er ønskelig samme med utseende
    og funksjonalitet på label og hjelpetekst som for de øvrige
    inputkomponentene, for eksempel seksjoner eller knapperader.
  </p>
  <p>
    Flytende utropsboks brukes som hovedregel i interne løsninger og i tabeller.
  </p>
</>
```

```js noeditor uu
<>
  <h3>Tips</h3>
  <ul>
    <li>
      Merk at flytende Callout (Calloutfloating=true) kun kan brukes i interne
      løsninger.
    </li>
  </ul>

  <h3>Mest relevante WCAG-krav</h3>
  <ul>
    <li>1.3.1 A, Informasjon og relasjoner</li>
    <li>4.1.2 A, Navn, rolle, verdi</li>
  </ul>

  <h3>WAI-ARIA</h3>
  <ul>
    <li>Aria-expanded brukes på knappene som utvides/minimeres.</li>
    <li>Aria-label brukes for å navngi lukk-ikon i dialogen.</li>
    <li>Aria-hidden brukes for å skjule ikoner for skjermleser. </li>
  </ul>
</>
```
