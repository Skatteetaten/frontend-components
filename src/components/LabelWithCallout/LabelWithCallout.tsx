import * as React from 'react';
import { getClassNames } from './LabelWithCallout.classNames';
import { IconButton } from '@fluentui/react';
import { Callout } from '../Callout';
import classnames from 'classnames';
import { calloutState, LabelWithCalloutProps } from './LabelWithCallout.types';

/**
 * @visibleName LabelWithCallout _MerkelappMedUtropsboks_
 */
export const LabelWithCallout = (props: LabelWithCalloutProps) => {
  const {
    ariaLabel,
    autoDismiss,
    border,
    buttonAriaLabel,
    buttonTitle,
    calloutFloating = false,
    className,
    editable,
    editFunction,
    help,
    id,
    inputId,
    inFieldset,
    label,
    readOnly,
    warning,
    onRenderLabel,
    onCalloutToggle,
  } = props;
  const styles = getClassNames({ calloutFloating, ...props });
  const [isCalloutVisible, setIsCalloutVisible] = React.useState(false);
  const [currentCalloutState, setCurrentCalloutState] = React.useState(
    calloutState.CLOSED
  );
  const iconButtonElementRef = React.useRef<HTMLSpanElement>(null);
  const helpElement = React.isValidElement(help) ? help : <p>{help}</p>;
  const warningElement = React.isValidElement(warning) ? (
    warning
  ) : (
    <p>{warning}</p>
  );

  const toggleEvent = () => {
    if (onCalloutToggle) {
      const oldCalloutState = currentCalloutState;
      const newCalloutState = isCalloutVisible
        ? calloutState.CLOSED
        : calloutState.OPEN;
      setCurrentCalloutState(newCalloutState);
      onCalloutToggle(oldCalloutState, newCalloutState);
    }
    return;
  };

  return onRenderLabel ? (
    onRenderLabel
  ) : (
    <>
      <div className={styles.calloutLabelWrapper}>
        {inFieldset ? (
          label ? (
            <legend
              aria-label={ariaLabel}
              id={id}
              className={classnames(styles.labelAsLegend, className)}
            >
              {label}
            </legend>
          ) : null
        ) : label ? (
          <label
            id={id}
            htmlFor={inputId}
            aria-label={ariaLabel}
            className={classnames(styles.label, className)}
          >
            {label}
          </label>
        ) : null}

        {help && !warning && (
          <span className={styles.labelIconArea} ref={iconButtonElementRef}>
            <IconButton
              iconProps={{ iconName: 'HelpOutline' }}
              title={buttonTitle ? buttonTitle : 'Hjelp'}
              aria-describedby={id}
              ariaLabel={buttonAriaLabel ? buttonAriaLabel : 'Hjelp'}
              aria-expanded={isCalloutVisible}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                toggleEvent();
              }}
              className={styles.icon}
            />
          </span>
        )}
        {warning && (
          <span className={styles.labelIconArea} ref={iconButtonElementRef}>
            <IconButton
              iconProps={{ iconName: 'WarningOutline' }}
              title={buttonTitle ? buttonTitle : 'Varsel'}
              aria-describedby={id}
              ariaLabel={buttonAriaLabel ? buttonAriaLabel : 'Varsel'}
              aria-expanded={isCalloutVisible}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                toggleEvent();
              }}
              className={styles.warningicon}
            />
          </span>
        )}
        {readOnly && (
          <span className={styles.labelIconArea}>
            {editable && (
              <IconButton
                iconProps={{ iconName: 'Edit' }}
                title={buttonTitle ? buttonTitle : 'Rediger'}
                aria-describedby={id}
                ariaLabel={buttonAriaLabel ? buttonAriaLabel : 'Rediger'}
                aria-expanded={isCalloutVisible}
                onClick={editFunction}
                className={styles.icon}
              />
            )}
          </span>
        )}
      </div>
      {isCalloutVisible && (
        <Callout
          className={styles.calloutContext}
          border={border}
          directionalHint={
            calloutFloating ? Callout.POS_BOTTOM_LEFT : Callout.POS_TOP_LEFT
          }
          color={help && !warning ? Callout.HELP : Callout.WARNING}
          target={iconButtonElementRef.current}
          onClose={() => {
            setIsCalloutVisible(false);
            toggleEvent();
          }}
          autoDismiss={autoDismiss}
          onDismiss={() => {
            if (autoDismiss) {
              setIsCalloutVisible(false);
              toggleEvent();
            }
          }}
        >
          {help && !warning ? helpElement : warningElement}
        </Callout>
      )}
    </>
  );

  /*  return onRenderLabel ? (
    onRenderLabel
  ) : inFieldset ? (
    <legend className={styles.labelArea}>{label}</legend>
  ) : (
    <div
      
      aria-label={ariaLabel}
      className={classnames(styles.labelArea, className)}
    >
      <span className={styles.label}>
        {label ? <Label>{label}</Label> : null}
      </span>
      
    </div>
  ); */
};
