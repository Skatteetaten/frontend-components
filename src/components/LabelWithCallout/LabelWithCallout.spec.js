import React from 'react';
import { mount, shallow } from 'enzyme';
import { matches } from './../utils/test-utils';
import LabelWithCallout from './LabelWithCallout';

function oppsettFullDOM(props) {
  return mount(<LabelWithCallout {...props} />);
}
function oppsettShallow(props) {
  return shallow(<LabelWithCallout {...props} />);
}

describe('LabelWithCallout komponent', () => {
  it('matcher snapshot', () => {
    matches(
      <LabelWithCallout
        id={'LabelWithCallout_id'}
        calloutFloating={false}
        help={'Dette er en mock-hjepetekst'}
        label={'Mock Label'}
      />
    );
  });
  it('rendrer LabelWithCallout med riktig props for hjelpetekst', () => {
    const wrapper = oppsettFullDOM({
      id: 'LabelWithCallout_id',
      calloutFloating: false,
      help: 'Dette er en mock-hjepetekst',
      label: 'Mock Label'
    });
    expect(
      wrapper
        .find('label')
        .first()
        .prop('id')
    ).toEqual('LabelWithCallout_id');
    const helpIcon = wrapper.find('.ms-Button--icon');
    const icon = helpIcon.find('i');
    expect(icon.prop('data-icon-name')).toEqual('HelpOutline');
    expect(helpIcon.prop('title')).toEqual('Hjelp');
    helpIcon.simulate('click');
    const callout = wrapper.find('.ms-Callout');
    expect(wrapper.exists('Callout')).toEqual(true);
    expect(callout.html()).toContain('Dette er en mock-hjepetekst');
  });
  it('rendrer LabelWithCallout med riktig props for varselstekst', () => {
    const wrapper = oppsettFullDOM({
      id: 'LabelWithCallout_id',
      calloutFloating: false,
      warning: 'Dette er en mock-varselstekst',
      label: 'Mock Label'
    });
    expect(
      wrapper
        .find('label')
        .first()
        .prop('id')
    ).toEqual('LabelWithCallout_id');
    const warningIcon = wrapper.find('.ms-Button--icon');
    const icon = warningIcon.find('i');
    expect(icon.prop('data-icon-name')).toEqual('WarningOutline');
    expect(warningIcon.prop('title')).toEqual('Varsel');
    warningIcon.simulate('click');
    const callout = wrapper.find('.ms-Callout');
    expect(wrapper.exists('Callout')).toEqual(true);
    expect(callout.html()).toContain('Dette er en mock-varselstekst');
  });
  it('lukker callout på kryss ut', () => {
    const wrapper = oppsettFullDOM({
      id: 'LabelWithCallout_id',
      calloutFloating: false,
      help: 'Dette er en mock-hjepetekst',
      label: 'Mock Label'
    });
    const helpIcon = wrapper.find('.ms-Button--icon');
    helpIcon.simulate('click');
    expect(wrapper.exists('Callout')).toEqual(true);
    const closeBtn = wrapper.find('.ms-Button--icon').last();
    closeBtn.simulate('click');
    expect(wrapper.exists('Callout')).toEqual(false);
  });
  it('skal kjøre brukerspesifikk event på click', () => {
    const mockFunksjon = jest.fn((gammel, ny) => gammel + ny);
    const wrapper = oppsettFullDOM({
      id: 'LabelWithCallout_id',
      calloutFloating: false,
      help: 'Dette er en mock-hjepetekst',
      label: 'Mock Label',
      onCalloutToggle: (gammel, ny) => mockFunksjon(gammel, ny)
    });
    const helpIcon = wrapper.find('.ms-Button--icon');
    expect(mockFunksjon).not.toHaveBeenCalled();
    helpIcon.simulate('click');
    expect(mockFunksjon).toHaveBeenCalled();
    expect(mockFunksjon.mock.calls[0]).toEqual(['CLOSED', 'OPEN']);
  });
  it('skal rendre legend dersom inFieldset er true', () => {
    const wrapper = oppsettShallow({
      id: 'LabelWithCallout_id',
      calloutFloating: false,
      help: 'Dette er en mock-hjepetekst',
      label: 'Mock Label',
      inFieldset: true
    });
    expect(wrapper.find('legend').exists()).toEqual(true);
    expect(wrapper.find('label').exists()).toEqual(false);
  });
});
