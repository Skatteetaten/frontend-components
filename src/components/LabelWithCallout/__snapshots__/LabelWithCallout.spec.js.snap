// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelWithCallout komponent matcher snapshot 1`] = `
Array [
  <label
    className=

        {
          color: #1d1d1d;
          display: initial;
          font-size: 1rem;
          font-weight: 400;
          padding-bottom: 4px;
          padding-left: 0px;
        }
        & .ms-Label {
          font-weight: 400;
        }
    id="LabelWithCallout_id"
  >
    Mock Label
  </label>,
  <span
    className=

        {
          display: inline-block;
          height: 22px;
        }
        & button  {
          height: 22px;
          margin-top: -2px;
        }
  >
    <button
      aria-describedby="LabelWithCallout_id"
      aria-expanded={false}
      aria-label="Hjelp"
      className=
          ms-Button
          ms-Button--icon
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            background-color: transparent;
            border-radius: 2px;
            border: none;
            box-sizing: border-box;
            color: #1362ae;
            cursor: pointer;
            display: inline-block;
            font-family: 'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif;
            font-size: 1rem;
            font-weight: 400;
            height: 32px;
            outline: transparent;
            padding-bottom: 0;
            padding-left: 4px;
            padding-right: 4px;
            padding-top: 0;
            position: relative;
            text-align: center;
            text-decoration: none;
            user-select: none;
            width: 32px;
          }
          &::-moz-focus-inner {
            border: 0;
          }
          .ms-Fabric--isFocusVisible &:focus:after {
            border: 1px solid transparent;
            bottom: 2px;
            content: "";
            left: 2px;
            outline: 1px solid rgba(55, 55, 55, 0.95);
            position: absolute;
            right: 2px;
            top: 2px;
            z-index: 1;
          }
          @media screen and (-ms-high-contrast: active), (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
            bottom: -2px;
            left: -2px;
            outline-color: ButtonText;
            right: -2px;
            top: -2px;
          }
          &:active > * {
            left: 0px;
            position: relative;
            top: 0px;
          }
          &:hover {
            background-color: #f4f4f4;
            color: #11599c;
          }
          @media screen and (-ms-high-contrast: active), (forced-colors: active){&:hover {
            border-color: Highlight;
            color: Highlight;
          }
          &:active {
            background-color: #eaeaea;
            color: #0c3f6f;
          }
          & i {
            font-size: large;
          }
          &:focus&:after {
            border: 2px solid #1362ae;
            outline: none;
          }
      data-is-focusable={true}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      title="Hjelp"
      type="button"
    >
      <span
        className=
            ms-Button-flexContainer
            {
              align-items: center;
              display: flex;
              flex-wrap: nowrap;
              height: 100%;
              justify-content: center;
            }
        data-automationid="splitbuttonprimary"
      >
        <i
          aria-hidden={true}
          className=
              ms-Icon
              ms-Button-icon
              {
                display: inline-block;
              }
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                font-family: "Material Icons";
                font-style: normal;
                font-weight: normal;
                speak: none;
              }
              {
                flex-shrink: 0;
                font-size: 1.0625rem;
                height: 16px;
                line-height: 16px;
                margin-bottom: 0;
                margin-left: 4px;
                margin-right: 4px;
                margin-top: 0;
                text-align: center;
              }
          data-icon-name="HelpOutline"
          style={
            Object {
              "fontFamily": "\\"Material Icons\\"",
            }
          }
        >
          ï˜¥
        </i>
      </span>
    </button>
  </span>,
]
`;
